### ISSUE TITLE
v3.34 backend_set lb_session_persistence

### ISSUE THREAD

**Issue by karljons** [terraform-providers/terraform-provider-oci#825]
We are trying to create a load balancer backend-set with the new lb_session_persistence_configuration block and it is not working.  Making same change through the cloud UI works so we think the issue is with the oci provider.

Old code:
  session_persistence_configuration {
    cookie_name = var.cookie_name
  }
  ssl_configuration {
    certificate_name        = var.certificate_name
    verify_peer_certificate = "false"
  }
}

New code:
      lb_cookie_session_persistence_configuration {
        cookie_name = var.cookie_name
     }

  ssl_configuration {
    certificate_name        = var.certificate_name
    verify_peer_certificate = "false"
  }
}

This code is in a submodule:
Root -> create cluster -> create load balancers -> create backend-set

We deleted the .terraform folder from the root folder and did a terraform init which brought the provider to 3.34

Execute plan - and no errors
Execute apply and fails with error:

Error: Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: 074f0093111352b74f915b25b71014cb/BEFFBA943C756047602EEE1649AD5AA8/4B8A7AEA5B3EF664C8DFBDCDF7C6E793

  on ../../../Modules/Manhattan/LBAAS-Config/BackendSet/main.tf line 28, in resource "oci_load_balancer_backend_set" "HTTPS-BACKEND-SET":
  28: resource "oci_load_balancer_backend_set" "HTTPS-BACKEND-SET" {

When I look at the plan output, I notice this:
      + lb_cookie_session_persistence_configuration {
          + cookie_name        = (known after apply)
          + disable_fallback   = (known after apply)
          + domain             = (known after apply)
          + is_http_only       = (known after apply)
          + is_secure          = (known after apply)
          + max_age_in_seconds = (known after apply)
          + path               = (known after apply)
        }

      + session_persistence_configuration {
          + cookie_name      = "X-Oracle-BMC-LBS-Route-qa2"
          + disable_fallback = false
        }

Looks like the provider is passing both lb_cookie_session and session_persistence configuration blocks to the API - these are mutually exclusive and I believe this is the root cause of the error.

Also, the cookie name that is in my code in the lb_cookie_session block is now in the session_persistence block, not in the lb block.

Expected output:
      + lb_cookie_session_persistence_configuration {
          + cookie_name        = "X-Oracle-BMC-LBS-Route-qa2"
          + disable_fallback   = (known after apply)
          + domain             = (known after apply)
          + is_http_only       = (known after apply)
          + is_secure          = (known after apply)
          + max_age_in_seconds = (known after apply)
          + path               = (known after apply)
        }

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version and Provider Version

Terraform v0.12.3
+ provider.oci v3.34.0
+ provider.vault v2.1.0

### Affected Resource(s)

oci_load_balancer_backend_set

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. 
# Please remove any sensitive information from configuration files before sharing them. 
```

### Steps to Reproduce

1. `terraform apply`

**Comment by kilokahn** [comment]
Thanks for reporting this @karljons . 

>Looks like the provider is passing both lb_cookie_session and session_persistence configuration blocks to the API - these are mutually exclusive and I believe this is the root cause of the error

If that was the case, the API would have returned a 400 InvalidParameter though. It does seem like a problem nonetheless. Can you please help out by providing the debug logs for the operation (using the instructions at https://www.terraform.io/docs/providers/oci/guides/troubleshooting.html#verbose-logging-for-oci-terraform-provider )? We will try to reproduce this on our side and also look for a workaround in the meantime.

**Comment by karljons** [comment]
I have a pretty steep deadline on this so trying to create it with regular session persistence by downgrading to v3.33 - I will reproduce as soon as I can take a break from getting these production instances up :)

**Comment by kilokahn** [comment]
@karljons we have identified a potential fix for cases when the attribute is being migrated from the old `session_persisitence_configuration` to the new `lb_cookie_session_persistence_configuration` field (which results in a 400 in our repro) that will be rolled out today. You can try it out after the release today. 
We are still awaiting logs from you to see if there is yet another 404 related problem that you were seeing in your case. Thanks again for reporting!

**Comment by kilokahn** [comment]
@karljons a fix for this case has been rolled out in v3.35.0. Please let us know if you still see any issues.

**Comment by kilokahn** [comment]
@karljons I am closing out this issue assuming that the underlying issue has been resolved. If that is not the case, please re-open the issue. Thanks!
